src/services/rfqm_service.ts

```
@@ -0,0 +1,136 @@
+// tslint:disable:max-file-line-count
+import { AssetSwapperContractAddresses, MarketOperation, ProtocolFeeUtils, QuoteRequestor } from '@0x/asset-swapper';
+import { BigNumber } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+
+import { META_TX_WORKER_REGISTRY, RFQT_REQUEST_MAX_RESPONSE_MS } from '../config';
+import { NULL_ADDRESS, ONE_SECOND_MS, RFQM_MINUMUM_EXPIRY_DURATION_MS } from '../constants';
+
+export interface FetchIndicativeQuoteParams {
+    apiKey: string;
+    buyAmount?: BigNumber;
+    buyToken: string;
+    buyTokenDecimals: number;
+    sellAmount?: BigNumber;
+    sellToken: string;
+    sellTokenDecimals: number;
+    takerAddress?: string;
+}
+
+export interface FetchIndicativeQuoteResponse {
+    allowanceTarget?: string;
+    buyAmount: BigNumber;
+    buyTokenAddress: string;
+    gas: BigNumber;
+    price: BigNumber;
+    sellAmount: BigNumber;
+    sellTokenAddress: string;
+}
+
+const RFQM_DEFAULT_OPTS = {
+    takerAddress: NULL_ADDRESS,
+    txOrigin: META_TX_WORKER_REGISTRY || NULL_ADDRESS,
```

As discussed, can we raise an error if no metatransaction registry is set?